explore/src/main/scala/explore/targeteditor/ElevationPlotSemester.scala

```diff
@@ -57,111 +58,108 @@ object ElevationPlotSemester:
                             }
-                            .map { case (_, maxVisibility) => maxVisibility }
-                      }
-                      .scan1(math.max)
-                      .last
-                      .evalMap(
-                        _.filterNot(_ > MinVisibility)
-                          .map(_ =>
+                        }
+                          .flatTap { case (maxInstant, _) =>
                             IO(
-                              xAxis.setTitle(XAxisTitleOptions().setText("Target is below horizon"))
+                              xAxis.addPlotLine(
+                                AxisPlotLinesOptions
+                                  .XAxisPlotLinesOptions()
+                                  .setId("progress")
+                                  .setValue(maxInstant)
+                                  .setZIndex(1000)
+                                  .setClassName("plot-plot-line-progress")
+                              )
                             ).void
-                          )
-                          .orEmpty
-                      ) ++
-                      fs2.Stream(xAxis.removePlotLine("progress"))
-                  )
-              }
-              .getOrElse(Resource.pure(fs2.Stream()))
-              .use(_.compile.drain)
+                          }
+                          .map { case (_, maxVisibility) => maxVisibility }
+                    }
+                    .scan1(math.max)
+                    .last
+                    .evalMap(
+                      _.filterNot(_ > MinVisibility)
+                        .map(_ =>
+                          IO(
+                            xAxis.setTitle(XAxisTitleOptions().setText("Target is below horizon"))
+                          ).void
+                        )
+                        .orEmpty
+                    ) ++
+                    fs2.Stream(xAxis.removePlotLine("progress"))
+                )
+            }
+            .getOrElse(Resource.pure(fs2.Stream()))
+            .use(_.compile.drain)
       )
-      .useEffectWithDepsBy((props, _, _, chartOpt) => (props.options.date, chartOpt))(
-        (props, _, _, _) =>
-          (date, chartOpt) =>
-            CallbackTo(chartOpt.value.value) >>=
-              (_.map(chart =>
-                Callback {
-                  // 2pm of the selected day, same as for semester start and end
-                  val localDateTime: LocalDateTime =
-                    LocalDateTime.of(date, LocalTime.MIDNIGHT).plusHours(14)
-                  val zonedDateTime: ZonedDateTime =
-                    ZonedDateTime.of(localDateTime, props.options.site.timezone)
+      .useEffectWithDepsBy((props, _, _, chartOpt) =>
+        (props.options.date, props.options.site, chartOpt)
+      )((_, _, _, _) =>
+        (date, site, chartOpt) =>
+          CallbackTo(chartOpt.value.value) >>=
```

Comment:
Why is this in a callback?

Reply:
I didn't change this, but since the whole body needs to be an effect, this was one way to do that.

Reply:
can it be just inlined in the next block?

Reply:
Yeah, it should be able to work like the `useEffect` above. I'll do that when I can get back on this branch.